src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala

```diff
@@ -560,13 +577,20 @@ object Helpers extends SparkSessionWrapper {
    * @return
    */
   def getWorkspaceByDatabase(
-                              etlDB: String,
+//                              etlDB: String,
+                              db: String,
                               organization_id: Option[String] = None,
                               apiUrl: Option[String] = None,
                               successfullOnly: Boolean = true,
                               disableValidations: Boolean = false
                             ): Workspace = {
     // verify database exists
+    val initialCatalog = spark.sessionState.catalogManager.currentCatalog.name()
+    val etlDB = if(db.contains(".")){
+      spark.sessionState.catalogManager.setCurrentCatalog(db.split("\\.").head)
```

Comment:
This code is very redundant throughout -- please create a Helper function like

setCatalog(spark, databaseName) = {...} and use this anywhere you need to use this. Imperative that you pass in the spark session though to ensure thread-local settings.

Reply:
done. created two different functions to get and set the catalog